client/my-sites/theme/main.jsx

```diff
@@ -465,75 +473,92 @@ const ThemeSheet = React.createClass( {
-	const { selectedSite: site, active: isActive, price, isLoggedIn } = stateProps;
+const ThemeSheetWithOptions = ( props ) => {
+	const { selectedSite: site, active: isActive, price, isLoggedIn } = props;
 
 	let defaultOption;
 
 	if ( ! isLoggedIn ) {
-		defaultOption = dispatchProps.signup;
+		defaultOption = 'signup';
 	} else if ( isActive ) {
-		defaultOption = dispatchProps.customize;
+		defaultOption = 'customize';
 	} else if ( price ) {
-		defaultOption = dispatchProps.purchase;
-		defaultOption.label = i18n.translate( 'Pick this design' );
+		defaultOption = 'purchase';
 	} else {
-		defaultOption = dispatchProps.activate;
-		defaultOption.label = i18n.translate( 'Activate this design' );
+		defaultOption = 'activate';
 	}
 
-	const dispatchOptions = {
-		defaultOption,
-		secondaryOption: dispatchProps.tryandcustomize
-	};
-
-	return Object.assign(
-		{},
-		ownProps,
-		stateProps,
-		site ? bindOptionsToSite( dispatchOptions, site ) : dispatchOptions,
+	return (
+		<ConnectedThemeSheet { ...props }
+			site={ site }
+			theme={ props /* TODO: Have connectOptions() only use theme ID */ }
+			options={ [
+				'signup',
+				'customize',
+				'tryandcustomize',
+				'purchase',
+				'activate'
+			] }
+			defaultOption={ defaultOption }
+			secondaryOption="tryandcustomize"
+			source="showcase-sheet" />
 	);
 };
 
 export default connect(
-	( state, props ) => {
+	// A number of the props that this mapStateToProps function computes are used
+	// by ThemeSheetWithOptions to compute defaultOption. After a state change
+	// triggered by an async action, connect()ed child components are, quite
+	// counter-intuitively, updated before their connect()ed parents (this is
+	// https://github.com/reactjs/redux/issues/1415), and might be fixed by
+	// react-redux 5.0.
+	// For this reason, after e.g. activating a theme in single-site mode,
+	// first the ThemeSheetWithOptions component's (child) connectOptions component
+	// will update in response to the currently displayed theme being activated.
+	// Doing so, it will filter and remove the activate option (adding customize
+	// instead). However, since the parent connect()ed-ThemeSheetWithOptions will
+	// only react to the state change afterwards, there is a brief moment when
+	// connectOptions still receives "activate" as its defaultOption prop, when
+	// activate is no longer part of its filtered options set, hence passing on
+	// undefined as the defaultOption object prop for its child. For the theme
+	// sheet, which eventually gets that defaultOption object prop, this means
+	// we must be careful to not accidentally access any attribute of that
+	// defaultOption prop. Otherwise, there will be an error that will prevent the
+	// state update from finishing properly, hence not updating defaultOption at all.
+	// The solution to this incredibly intricate issue is simple: Give ThemeSheet
+	// a valid defaultProp for defaultOption.
```

Comment:
Part of it is quite general (the stuff about this being a `react-redux` quirk), but since the rest is really quite specific to the code here (much more than general README notes), I'm inclined to leave it here. Will turn it into a multi-line `/* comment */` tho.